src/presign/participant.rs

```diff
@@ -1188,19 +1195,23 @@ impl PresignKeyShareAndInfo {
                 .aux_info_private
                 .decryption_key()
                 .decrypt(&r2_pub_j.D)
-                .map_err(|_| InternalError::InternalInvariantFailed)?;
+                .map_err(|_| InternalError::ProtocolError)?;
             let alpha_hat = self
                 .aux_info_private
                 .decryption_key()
                 .decrypt(&r2_pub_j.D_hat)
-                .map_err(|_| InternalError::InternalInvariantFailed)?;
+                .map_err(|_| InternalError::ProtocolError)?;
```

Comment:
todo: Can you add a log to these? I think the problem in both cases is that the value received from the given participant was out of range (the only way decryption fails is if the ciphertext is not in the range `(0, N)` -- a rough approximation of the multiplicative group of integers mod `N`).